contracts/account/extensions/draft-AccountERC7579Hooked.sol

```diff
@@ -0,0 +1,63 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.24;
+
+import {IERC7579Hook, MODULE_TYPE_HOOK, IERC7579ModuleConfig} from "../../interfaces/draft-IERC7579.sol";
+import {ERC7579Utils, Mode} from "../utils/draft-ERC7579Utils.sol";
+import {AccountERC7579} from "../draft-AccountERC7579.sol";
+
+abstract contract AccountERC7579Hooked is AccountERC7579 {
+    address private _hook;
+
+    modifier withHook() {
+        address hook_ = hook();
+        if (hook_ == address(0)) {
+            _;
+        } else {
+            bytes memory hookData = IERC7579Hook(hook_).preCheck(msg.sender, msg.value, msg.data);
+            _;
+            IERC7579Hook(hook_).postCheck(hookData);
+        }
+    }
```

Comment:
AFAIK, using `_;` twice will cause a lot of bytecode duplication. My understanding is that this would be much better for deployment size.

```suggestion
    modifier withHook() {
        bytes memory hookData;
        address hook_ = hook();
        if (hook_ != address(0)) hookData = IERC7579Hook(hook_).preCheck(msg.sender, msg.value, msg.data);
        _;
        if (hook_ != address(0)) IERC7579Hook(hook_).postCheck(hookData);
    }
```